### DISCUSSION TITLE
Display distance to nearest signal/junction when placing a signal

### DISCUSSION THREAD

**Discussion by daiwai** [voxeltycoon/issues#412] [UI&UX]
When placing a new signal, it would be nice to easily see the distance to the nearest signal or junction in both directions of the new signal. This would help when wanting to enforce a certain distance between signals in your network.

It could look something like this:
![signal_distance_overlay](https://user-images.githubusercontent.com/683371/90338740-1113af00-dfec-11ea-92d7-e50399aa8823.png)

**Comment by victorleblais** [comment]
Very good suggestion, and nice mockup.
I'm not sure your design is easy to do and/or easy to read in some cases (on curves, or when there are tracks or other objects "in the way" of your "tooltip").
Worst case scenario, it could be added in the price tooltip...

**Comment by SMERKIN5000** [comment]
This is something that would greatly improve the user experience.

**Comment by IDragonfire** [comment]
I had the same idea.

Use cases: Hard to see if the rail block is long enough to fit the longest train inside.

Also very useful for stations.... 
Train was 126m, station/rail block was 125m. 
It took me a while to debug why the block was occupied.

PS.: If it cost to much to implement the display while placing, at least show it for the signals that were build.

**Comment by IDragonfire** [comment]
Found a mod:
https://steamcommunity.com/sharedfiles/filedetails/?id=2770538697
